backend/internal/controller/server/msg_handler.go

```diff
@@ -0,0 +1,65 @@
+package server
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/HackYourCareer/SmartKickers/internal/controller/adapter"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	messageText = 1
+)
+
+func (s server) TableMessagesHandler(w http.ResponseWriter, r *http.Request) {
+	var upgrader websocket.Upgrader
+	c, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Println(err)
+	}
+
+	defer c.Close()
+
+	for {
+		_, receivedMsg, err := c.NextReader()
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+		response, err := s.createResponse(receivedMsg)
+
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+		if response != nil {
+			err = c.WriteMessage(messageText, response)
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+		}
+	}
+}
+
+func (s server) createResponse(reader io.Reader) ([]byte, error) {
```

Comment:
You could also take `writer` as second parameter and write directly to it, but one could argue that your current solution is cleaner to read

Reply:
In our opinion it is cleaner to read so we leave it as it is.